pkg/trustedresources/verify.go

```diff
@@ -168,11 +238,37 @@ func getVerifiers(ctx context.Context, k8s kubernetes.Interface) ([]signature.Ve
 			verifiers = append(verifiers, v...)
 		}
 	}
+
+	return verifiers
+}
+
+// verifiersFromPolicy get all verifiers from VerificationPolicy.
+// For each policy, loop the Authorities of the VerificationPolicy to fetch public key
+// from either inline Data or from a SecretRef.
+func verifiersFromPolicy(ctx context.Context, k8s kubernetes.Interface, policy *v1alpha1.VerificationPolicy) ([]signature.Verifier, error) {
+	verifiers := []signature.Verifier{}
+	for _, a := range policy.Spec.Authorities {
+		algorithm, err := hashAlgorithm(a.Key.HashAlgorithm)
+		if err != nil {
+			return nil, fmt.Errorf("authority %q contains an invalid hash algorithm: %w", a.Name, err)
+		}
+		if a.Key.Data != "" {
+			verifier, err := verifierFromKeyData([]byte(a.Key.Data), algorithm)
+			if err == nil {
+				verifiers = append(verifiers, verifier)
+			}
+		} else if a.Key.SecretRef != nil {
```

Comment:
what happens if both the data and the secret ref are blank? are we assuming that is caught via validation already? (if so maybe include a comment to explain? but im thinking even if we think validation will catch that it would make sense to return an error here just in case it doesnt)

Reply:
yes you're correct that it is caught by validation webhook, I can add a error check here